services/src/main/scala/cromwell/services/metadata/WorkflowQueryKey.scala

```diff
@@ -182,3 +188,19 @@ sealed trait IntWorkflowQueryKey extends WorkflowQueryKey[Option[Int]] {
   def displayName: String
 }
 
+sealed trait BooleanWorkflowQueryKey[A] extends WorkflowQueryKey[Boolean] {
+  override def validate(grouped: Map[String, Seq[(String, String)]]): ErrorOr[Boolean] = {
+    valuesFromMap(grouped) match {
+      case vs if vs.lengthCompare(1) > 0 => s"Found ${vs.size} values for key '$name' but at most one is allowed.".invalidNel[Boolean]
+      case Nil => true.validNel
+      case v :: Nil => {
```

Comment:
Be careful here - I think `valuesFromMap` will give you a `Seq` whereas (I believe) `:: Nil` only applies to `List` (which is a subtype of `Seq`, so this will work *most of the time*).

I think it'll be safer to, eg:
```scala
case vs if vs.size == 1 => 
  val v = vs.head 
  ...
```

Reply:
"most of the time" - we have definitely had bugs over the years of this nature, so emphasis on "most". Someone uses the cons operator in a pattern match and someone else passes a non-`List` `Seq` into the function, which is itself specifying a `Seq` input.

Reply:
Oh, I didn't know! Thanks!

Will it then make sense to change similar logic inside DateTimeWorkflowQueryKey and IntWorkflowQueryKey as well?

Reply:
I converted it to List and kept the original `v :: Nil`, so it's cleaner and consistent with similar code in other traits.